starport-core/src/main/scala/com/krux/starport/StartScheduledPipelines.scala

```diff
@@ -198,6 +104,18 @@ object StartScheduledPipelines extends StarportActivity {
     val actualStart = options.actualStart
     db.run(DBIO.seq(SchedulerMetrics() += SchedulerMetric(actualStart))).waitForResult
 
+    // in case of default taskDispatcher: the dispatchedPipelines and succesfulPipelines metrics should be exactly same.
+    val dispatchedPipelines = metrics.register("counter.successful-pipeline-dispatch-count", new Counter())
+    val successfulPipelines = metrics.register("counter.successful-pipeline-deployment-count", new Counter())
+    val failedPipelines = metrics.register("counter.failed-pipeline-deployment-count", new Counter())
+
+    val taskDispatcher: TaskDispatcher = conf.dispatcherType match {
+      case "default" => new TaskDispatcherImpl()
+      case x =>
+        // pipelines scheduled in a previous run should be fetched here if the dispatcher is remote
+        throw new NotImplementedError(s"there is no task dispatcher implementation for $x")
+    }
```

Comment:
@realstraw - As you had pointed out we should solve the problem of updating the scheduled pipelines before fetching the new ones to be scheduled in case of a remote dispatcher implementation.

Here I have moved the dispatcher and metrics initialization to *before* we fetch the new pipelines to be scheduled. Do you think this will suffice for now?